Source/WebKit/NetworkProcess/ServiceWorker/ServiceWorkerFetchTask.cpp

```diff
@@ -254,8 +263,29 @@ void ServiceWorkerFetchTask::didReceiveData(const IPC::SharedBufferReference& da
     if (m_isDone)
         return;
 
+    RefPtr<WebCore::SharedBuffer> buffer = data.unsafeBuffer();
+#if ENABLE(CONTENT_FILTERING_IN_NETWORKING_PROCESS)
+    if (buffer && !m_loader.continueAfterDataReceived(*buffer, encodedDataLength))
+        return;
+#endif
+
+    ASSERT(!m_timeoutTimer || !m_timeoutTimer->isActive());
+    sendToClient(Messages::WebResourceLoader::DidReceiveData { IPC::SharedBufferReference(buffer), encodedDataLength });
+}
+
+void ServiceWorkerFetchTask::didReceiveDataFromPreloader(const WebCore::FragmentedSharedBuffer& data, uint64_t encodedDataLength)
```

Comment:
Why do we need to break down `ServiceWorkerFetchTask::didReceiveData` into two functions? Is it because the buffer needs to be contiguous? `SharedBufferReference::unsafeBuffer()` seems can do that.

Reply:
> Why do we need to break down `ServiceWorkerFetchTask::didReceiveData` into two functions? Is it because the buffer needs to be contiguous? `SharedBufferReference::unsafeBuffer()` seems can do that.

The didReceiveData method in ServiceWorkerFetchTask has been split into two methods, since it was being called both when receiving data over IPC and from the network. The unsafeBuffer method in IPC::SharedBufferReference is only allowed to be called on the receiver side.